### PR TITLE
Add vector arithmetic operations and validation

### LINKED ISSUE #10: [Feature Request] vector needs numerical functions
**Is your feature request related to a problem? Please describe.**
vector class does not has any numerical functions.

**Describe the solution you'd like**
add numerical functions such that dot, plus, etc.

### PR DESCRIPTION
**Pull request by daisuke-nagao**
- Implemented addition, subtraction, scaling, and division operations for the vector class.
- Added operator overloads for +, -, *, and /.
- Introduced map function for element-wise operations.
- Added validation for size mismatch in arithmetic operations.
- Enhanced unit tests to cover new arithmetic functionalities and edge cases.

fix #10

### CODE REVIEW

File: include/dicek/linalg/vector.hpp [hunk 1]
```diff
@@ -329,13 +332,142 @@ class vector {
...
+  }
+
+  vector operator/(scalar_type val) const {
+    return map([val](scalar_type x) { return x / val; });
+  }
+
+  vector& operator+=(const vector& rhs) {
+    validate_same_size(rhs, "vector::operator+=");
+    for (std::size_t i = 0; i < size(); ++i) {
+      (*this)[i] += rhs[i];
```
[the remark this answers is not in the document]
Comment (daisuke-nagao) [reply include/dicek/linalg/vector.hpp]: @codex
